src/jit/importer.cpp

```diff
@@ -7664,6 +7676,21 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
             }
             else
             {
+                if (fatPointerCandidate)
+                {
+                    assert(!bIntrinsicImported);
+                    assert(IsTargetAbi(CORINFO_CORERT_ABI));
+                    if (call->OperGet() != GT_LCL_VAR) // can be already converted by impFixupCallStructReturn.
+                    {
+                        unsigned   calliSlot  = lvaGrabTemp(true DEBUGARG("calli"));
+                        LclVarDsc* varDsc     = &lvaTable[calliSlot];
+                        varDsc->lvVerTypeInfo = tiRetVal;
+                        impAssignTempGen(calliSlot, call, clsHnd, (unsigned)CHECK_SPILL_NONE);
```

Comment:
Looks like that callRetType cannot be void because of the check on line 7605.

Also impFixupCallStructReturn() will not modify struct return type of a call.

---
In reply to: [99204735](https://github.com/dotnet/coreclr/pull/9191#discussion_r99204735) [](ancestors = 99204735)